pandas-stubs/_typing.pyi

```diff
@@ -203,7 +204,20 @@ XMLParsers = Literal["lxml", "etree"]
 # Any plain Python or numpy function
 Function = Union[np.ufunc, Callable[..., Any]]
 GroupByObjectNonScalar = Union[
-    tuple, list[HashableT], Function, Series, np.ndarray, Mapping[Label, Any], Index
+    tuple,
+    list[HashableT],
```

Comment:
Asiide from list[HashableT] these are not in the doc string.  Looks like they would, but would be best to update the pandas docs first to clarify that list of functions, arrays, etc are accepted.

Reply:
They are in the newest docs here: https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.groupby.html